File: app/src/main/java/net/emilla/util/MediaControl.java
package net.emilla.util;

import static android.view.KeyEvent.ACTION_DOWN;
import static android.view.KeyEvent.ACTION_UP;

import android.media.AudioManager;
import android.view.KeyEvent;

public enum MediaControl {
    ;

    public static void play(AudioManager am) {
        sendButtonEvent(am, KeyEvent.KEYCODE_MEDIA_PLAY);
    }

    public static void pause(AudioManager am) {
        sendButtonEvent(am, KeyEvent.KEYCODE_MEDIA_PAUSE);
    }

    public static void playPause(AudioManager am) {
        sendButtonEvent(am, KeyEvent.KEYCODE_MEDIA_PLAY_PAUSE);
    }

    private static void sendButtonEvent(AudioManager am, int keyCode) {
        am.dispatchMediaKeyEvent(new KeyEvent(ACTION_DOWN, keyCode));
        am.dispatchMediaKeyEvent(new KeyEvent(ACTION_UP, keyCode));
    }

}

File: app/src/main/java/net/emilla/util/MediaControl.kt
@file:JvmName("MediaControl")

package net.emilla.util

import android.media.AudioManager
import android.view.KeyEvent

fun AudioManager.sendPlayEvent() = sendButtonEvent(KeyEvent.KEYCODE_MEDIA_PLAY)
fun AudioManager.sendPauseEvent() = sendButtonEvent(KeyEvent.KEYCODE_MEDIA_PAUSE)
fun AudioManager.sendPlayPauseEvent() = sendButtonEvent(KeyEvent.KEYCODE_MEDIA_PLAY_PAUSE)

private fun AudioManager.sendButtonEvent(keyCode: Int) {
    dispatchMediaKeyEvent(KeyEvent(KeyEvent.ACTION_DOWN, keyCode))
    dispatchMediaKeyEvent(KeyEvent(KeyEvent.ACTION_UP, keyCode))
}
